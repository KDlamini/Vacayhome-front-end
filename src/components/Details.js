/* eslint-disable camelcase */
/* eslint-disable react/prop-types */
import React from 'react';
import { BiArrowBack } from 'react-icons/bi';

const Detail = (props) => {
  console.log(props);
  const {
    image_url, name, description, price, location,
  } = props;
  return (
    <>
      <div>
        <img src={image_url} alt="TRopical houses" />
      </div>
      <div>
        <h3>{name}</h3>
        <p>{description}</p>
      </div>
      <div>
        <p>
          This vacation home is located in
          {' '}
          {location}
        </p>
        <p>
          For the price of
          {' '}
          $
          {price}
        </p>
      </div>
      <div>
        <button type="button">Reserve</button>
      </div>
      <div>
        <a href="/">
          <BiArrowBack />
        </a>
      </div>
    </>
  );
};

export default Detail;
