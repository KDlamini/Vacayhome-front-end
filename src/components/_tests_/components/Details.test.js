import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../redux/configureStore';
import Details from '../../Details';

describe('BookAppointment', () => {
  it('renders correctly', () => {
    const props = {
      id: 1,
      name: 'house1',
      price: '100.0',
      description: 'This is house 1',
      country: 'US',
      city: 'Los Angeles',
      address: '100 cool street',
      property_type: 'house',
      image_url: 'https://images.pexels.com/photos/106399/pexels-photo-106399.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500',
      created_at: '2022-04-05T13:13:11.814Z',
      updated_at: '2022-04-05T13:13:11.814Z',
    };

    window.localStorage.setItem('userId', 1);
    const tree = render(
      <Provider store={store}><Router><Details props={props} /></Router></Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
