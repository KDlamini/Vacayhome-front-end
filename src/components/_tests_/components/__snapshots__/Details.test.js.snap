// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid"
      >
        <div
          class="homepage-container bg-white row"
        >
          <div
            class="details-container"
          >
            <div
              class="flex-direction"
            >
              <div>
                <img
                  alt="vacation house"
                  class="thumbnail images rounded"
                  src="https://images.pexels.com/photos/106399/pexels-photo-106399.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
                />
              </div>
              <div
                class="card-body"
              >
                <div
                  class="details-text-container"
                >
                  <div>
                    <h3
                      class="text-dark"
                    >
                      house1
                    </h3>
                    <p
                      class="description my-4"
                    >
                      This is house 1
                    </p>
                    <div>
                      <p
                        class="description"
                      >
                        This vacation home is located in
                         
                        100 cool street
                        ,  
                        Los Angeles
                        ,  
                        US
                      </p>
                      <p
                        class="description"
                      >
                        For the price of
                         
                        $
                        100.0
                      </p>
                    </div>
                    <div>
                      <a
                        href="/null/1/reservation"
                      >
                        <button
                          class="btn-success theme-btn mt-4 rounded"
                          type="button"
                          variant="success"
                        >
                          Reserve
                        </button>
                      </a>
                    </div>
                    <div>
                      <a
                        href="/null"
                      >
                        <svg
                          class="mt-4"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M21 11H6.414l5.293-5.293-1.414-1.414L2.586 12l7.707 7.707 1.414-1.414L6.414 13H21z"
                          />
                        </svg>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid"
    >
      <div
        class="homepage-container bg-white row"
      >
        <div
          class="details-container"
        >
          <div
            class="flex-direction"
          >
            <div>
              <img
                alt="vacation house"
                class="thumbnail images rounded"
                src="https://images.pexels.com/photos/106399/pexels-photo-106399.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
              />
            </div>
            <div
              class="card-body"
            >
              <div
                class="details-text-container"
              >
                <div>
                  <h3
                    class="text-dark"
                  >
                    house1
                  </h3>
                  <p
                    class="description my-4"
                  >
                    This is house 1
                  </p>
                  <div>
                    <p
                      class="description"
                    >
                      This vacation home is located in
                       
                      100 cool street
                      ,  
                      Los Angeles
                      ,  
                      US
                    </p>
                    <p
                      class="description"
                    >
                      For the price of
                       
                      $
                      100.0
                    </p>
                  </div>
                  <div>
                    <a
                      href="/null/1/reservation"
                    >
                      <button
                        class="btn-success theme-btn mt-4 rounded"
                        type="button"
                        variant="success"
                      >
                        Reserve
                      </button>
                    </a>
                  </div>
                  <div>
                    <a
                      href="/null"
                    >
                      <svg
                        class="mt-4"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M21 11H6.414l5.293-5.293-1.414-1.414L2.586 12l7.707 7.707 1.414-1.414L6.414 13H21z"
                        />
                      </svg>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
